hal/armv7m/imxrt/10xx/console.c

```diff
@@ -15,6 +15,23 @@
 
 #include <hal/hal.h>
 
+#define CONCAT(a, b)         a##b
+#define CONCAT2(a, b)        CONCAT(a, b)
+#define UART_PIN(n, pin)     CONCAT(UART, n##_##pin)
+#define CONSOLE_BASE(n)      (UART_PIN(n, BASE))
+#define CONSOLE_CLK(n)       (UART_PIN(n, CLK))
+#define CONSOLE_MUX(n, pin)  (CONCAT2(pctl_mux_gpio_, UART_PIN(n, pin)))
+#define CONSOLE_PAD(n, pin)  (CONCAT2(pctl_pad_gpio_, UART_PIN(n, pin)))
+#define CONSOLE_ISEL(n, pin) (CONCAT2(pctl_isel_lpuart, CONCAT(n, _##pin)))
+#define CONSOLE_BAUD(n)      (UART_PIN(n, BAUDRATE))
+
+#if CONSOLE_BAUD(UART_CONSOLE)
+#define CONSOLE_BAUDRATE CONSOLE_BAUD(UART_CONSOLE)
```

Comment:
`CONSOLE_BAUDRATE` is already in use in phoenix-rtos-kernel:
https://github.com/phoenix-rtos/phoenix-rtos-kernel/blob/071fe90ba4373afedad6296db12d2417669299bd/hal/armv7m/imxrt/10xx/console.c#L21
I use it in `BOARD_CONFIG` in order to set the kernel early console baudrate, so your code triggers a warning in this case. However, if you add the same mechanism to the kernel this `CONSOLE_BAUDRATE` in `BOARD_CONFIG` won't be needed, so there won't be a collision. Therefore, it's fine for me.

Reply:
Yes, it will work like this: any of the `n` uarts can define a custom `UARTn_BAUDRATE` in `BOARD_CONFIG` instead of a single `CONSOLE_BAUDRATE`, this is now consistent with `imxrt-multi` and `plo` (current PR), so the `kernel`s early console is underway  (I'm testing it now). That's right, I changed the logic in the kernel in a similar way to this PR.